zerver/tests/fixtures/katex_test_cases.json

```diff
@@ -0,0 +1,36 @@
+            "expected_output": "intro text\n$$p+q =4$$\n$$x+y=z$$\nhello world"
+        },
+        {
+            "description": "Inline expression with intermediate text node between two inline expressions.",
+            "input": "<p>hey there<br>\n<span class=\"katex\"><span class=\"katex-mathml\"><math xmlns=\"http://www.w3.org/1998/Math/MathML\"><semantics><mrow><mi>a</mi><mo>+</mo><mi>b</mi></mrow><annotation encoding=\"application/x-tex\">a+b</annotation></semantics></math></span><span class=\"katex-html\" aria-hidden=\"true\"><span class=\"base\"><span class=\"strut\" style=\"height:0.6667em;vertical-align:-0.0833em;\"></span><span class=\"mord mathnormal\">a</span><span class=\"mspace\" style=\"margin-right:0.2222em;\"></span><span class=\"mbin\">+</span><span class=\"mspace\" style=\"margin-right:0.2222em;\"></span></span><span class=\"base\"><span class=\"strut\" style=\"height:0.6944em;\"></span><span class=\"mord mathnormal\">b</span></span></span></span> some text <span class=\"katex\"><span class=\"katex-mathml\"><math xmlns=\"http://www.w3.org/1998/Math/MathML\"><semantics><mrow><mi>c</mi><mo>+</mo><mi>d</mi></mrow><annotation encoding=\"application/x-tex\">c+d</annotation></semantics></math></span><span class=\"katex-html\" aria-hidden=\"true\"><span class=\"base\"><span class=\"strut\" style=\"height:0.6667em;vertical-align:-0.0833em;\"></span><span class=\"mord mathnormal\">c</span><span class=\"mspace\" style=\"margin-right:0.2222em;\"></span><span class=\"mbin\">+</span><span class=\"mspace\" style=\"margin-right:0.2222em;\"></span></span><span class=\"base\"><span class=\"strut\" style=\"height:0.6944em;\"></span><span class=\"mord mathnormal\">d</span></span></span></span><br>\nhow </p>",
+            "expected_output": "hey there\n$$a+b$$ some text $$c+d$$\nhow"
```

Comment:
We need some tests that don't have newlines in the input in between expressions, since that's very common in real communication. 

And it seems to not work. If I paste this inline expression into a new Zulip message: `hey there $$a+b$$ some text $$a+b$$ how`, and then copy it and paste into the compose box, what I get out is `hey there hey there $$a+b$$ some text $$a+b$$ how some text  how`, i.e. the text content are weirdly duplicated in multiple places.

Reply:
Yeah, I think this framework is fine and we can try to figure out what's happening with that problematic case as a follow-up.